### PR TITLE
Headless mode

### PR DESCRIPTION
**Pull request by byronz**
this solves  #793 

- the headless mode option is added so you can do `pytest --headless True`, default is False
- it might be useful for advanced testers to further configure the selenium options via a hook. the current solution opens the options object, I feel it's a good compromise here.  But it might be good to know if users wanna full control on the webdriver construction. 

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] add headless mode
   -  [x] add hooks to configure webdriver options
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))

### optionals

- [x] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follow
    -  [ ] this github [#PR number]() updates the dash docs

### CODE REVIEW

**Comment by byronz** [comment]
@mmartinsky feel free to add comments on the proposed solution for headless mode. 

@alexcjohnson  we can probably switch all integration with headless mode too, it's faster.  I will test more

**Comment by mmartinsky** [comment]
Thanks @byronz ! Looking forward to release - do you happen to have any idea on release cadence post 1.0.0?

File: dash/testing/browser.py [hunk 1]
```diff
@@ -244,10 +252,20 @@ def get_webdriver(self, remote):
             )
         )
 
-    @staticmethod
-    def _get_chrome():
-        options = Options()
-        options.add_argument("--no-sandbox")
```
Comment (alexcjohnson) [inline dash/testing/browser.py]: Looks like we lost `--no-sandbox`, is it important?

File: dash/testing/plugin.py [hunk 2]
```diff
@@ -25,6 +25,25 @@ def pytest_addoption(parser):
         help="Name of the selenium driver to use",
     )
 
+    dash.addoption(
+        "--headless",
+        action="store",
+        default=False,
+        help="Run tests in headless mode",
+    )
```
Comment (alexcjohnson) [inline dash/testing/plugin.py]: could we use `"store_true"` so you can simplify to `pytest --headless`, rather than `pytest --headless True`?

Comment (byronz) [reply dash/testing/plugin.py]: yep, I like the shorter version

File: dash/testing/browser.py (hunk 1 above)
Comment (byronz) [reply dash/testing/browser.py]: no, I removed this part as the options are fully configurable after this PR

**Comment by byronz** [comment]
> Thanks @byronz ! Looking forward to release - do you happen to have any idea on release cadence post 1.0.0?

@mmartinsky  we have some internal discussion right now, nothing official yet, but 2-6 weeks, might be less frequent than last few months.

**Comment by mmartinsky** [comment]
hey @byronz and @alexcjohnson, do you have an example of altering the options for the webdriver? For background, I'm running into `DevToolsActivePort file doesn't exist.` when running in headless on Gitlab CI. 

I found https://github.com/heroku/heroku-buildpack-google-chrome/issues/46#issuecomment-396783086 that seems to be the same issue, I'm hoping that adding these options will clear this up.

The command I'm running is `pytest --headless`, nothing fancy there. Works like a charm locally.

Our docker image is using Chrome v.`67.0.3396.99-1`, chromedriver v`2.40`.

**Comment by alexcjohnson** [comment]
I'll let @byronz comment on using the options, but I'll note a (sporadic) `DevToolsActivePort file doesn't exist.` is exactly why we added an auto-retry of starting the browser, most recent version https://github.com/plotly/dash/pull/822#discussion_r303459530 (included in dash 1.0.2)

If you're seeing a consistent failure, the options may be necessary; if it's sporadic the auto-retry may fix it.

**Comment by byronz** [comment]
@mmartinsky good point about the how-to, I added the issue for doc enhancement.

a sample hook, I put this in a `conftest.py` file under a submodule of `tests` folder
```python
from selenium.webdriver.chrome.options import Options

def pytest_setup_options():
    """called before webdriver is initialized"""
    options = Options()
    options.add_argument('--no-sandbox')
    options.add_argument('--disable-dev-shm-usage')
    return options
```
and you can verify the options by increasing the logging level `pytest -k dvui001 --log-cli-level DEBUG`

```bash
Test session starts (platform: darwin, Python 3.7.4, pytest 4.6.4, pytest-sugar 0.9.2)
cachedir: .pytest_cache
rootdir: /Users/byron/code/dash, inifile: pytest.ini
plugins: sugar-0.9.2, mock-1.10.4, dash-1.0.1
collecting ...
------------------------------------------- live log setup --------------------------------------------
22:49:19 | DEBUG | selenium.webdriver.remote.remote_connection:388 | POST http://127.0.0.1:49395/session {"capabilities": {"firstMatch": [{}], "alwaysMatch": {"browserName": "chrome", "platformName": "any", "goog:chromeOptions": {"prefs": {"download.default_directory": "/private/var/folders/fl/dx4zhxfn11n3bbrrd7_hwhdr0000gn/T/pytest-of-byron/pytest-36/test_dvui001_disable_props_che0/download", "download.prompt_for_download": false, "download.directory_upgrade": true, "safebrowsing.enabled": false, "safebrowsing.disable_download_protection": true}, "extensions": [], "args": ["--no-sandbox", "--disable-dev-shm-usage"]}}}, "desiredCapabilities": {"browserName": "chrome", "version": "", "platform": "ANY", "loggingPrefs": {"browser": "SEVERE"}, "goog:chromeOptions": {"prefs": {"download.default_directory": "/private/var/folders/fl/dx4zhxfn11n3bbrrd7_hwhdr0000gn/T/pytest-of-byron/pytest-36/test_dvui001_disable_props_che0/download", "download.prompt_for_download": false, "download.directory_upgrade": true, "safebrowsing.enabled": false, "safebrowsing.disable_download_protection": true}, "extensions": [], "args": ["--no-sandbox", "--disable-dev-shm-usage"]}}}

```

we can tell that `"args": ["--no-sandbox", "--disable-dev-shm-usage"]}}}` is effective in selenium as configured in the hooks

**Pull request merged**